### ISSUE TITLE
Separatel seasons-related code into extension

### ISSUE THREAD

**Issue by carstingaxion** [slimndap/wp-theatre#323]
## Proposed NEW data modell (maybe version 1.0.0)

| `taxonomy` :arrow_right:<br/><br/>`post_type` :arrow_down:  | `post_tag` | `category` | *NEW* `wp_theatre_season_shadow` |
|--------|--------|--------|--------|
| `wp_theatre_prod` | :heavy_check_mark:  | :heavy_check_mark:   | :heavy_check_mark: | 
| `wp_theatre_event` | :knot:  | :knot:   | :heavy_check_mark: | 
| `wp_theatre_season` | :heavy_minus_sign:  |  :heavy_minus_sign: | :arrows_clockwise: | 

> [All this](https://github.com/slimndap/wp-theatre/issues/318#issuecomment-1828678394) could also be done for *Seasons*, but I suggest separating all seasons-related code into an extension. 

*Everytime I created websites for theater-companies or solo-artists, without an own physical stage, seasons were not needed at all. For all houses with own and/or guest artists, seasons were a strict need. It's a 50:50 thing, if website-owners need seasons-handling.*

When we have a working plan for [how events relate to productions](https://github.com/slimndap/wp-theatre/issues/318#issuecomment-1828678394), we can go on here.

**Comment by slimndap** [comment]
Agreed. Seasons are hardly being used in my experience.  Some theaters usesit to create a hidden page with all events for the next season. This allows them to preview all event contents, without publishing it on the website.
